### PR TITLE
IOS L3 interfaces resource model with supported operations

### PR DESCRIPTION
**Pull request by justjais**
Resource Model for ios_l3_interfaces module.

### CODE REVIEW

**Comment by cidrblock** [comment]
We will need to account for a `secondary` address:
```
Current configuration : 145 bytes
!
interface GigabitEthernet3
 ip address 10.0.0.1 255.255.255.0 secondary
 ip address 192.168.1.1 255.255.255.0
 shutdown
 negotiation auto
end

```

**Comment by cidrblock** [comment]
I'd like to get some feedback, but's here is one possible way to add the secondary into the model:

```
ios_interfaces:
   config:
     - name: GigabitEthernet0/2
       ipv4:
         address: 192.168.0.1/24
         secondary: 10.1.1.1/24
```

**Comment by cidrblock** [comment]
Or another possibility:
```
name: Merge provided configuration with device configuration
ios_interfaces:
  config:
    - name: GigabitEthernet0/2
      ip:
        addresses: 
          v4:
            primary: 192.168.0.1/24
            secondary: 10.1.1.1/24
```

File: ios/l3_interfaces/deleted_example_01.txt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#  ip address 192.168.1.0 255.255.255.0
+# interface GigabitEthernet0/3
+#  description Configured by Ansible Network
+#  ip address 192.168.0.1 255.255.255.0
+#  shutdown
+#  duplex full
+#  speed 10
+#  ipv6 address FD5D:12C9:2201:1::1/64
+# interface GigabitEthernet0/3.100
+#  encapsulation dot1Q 20
```
Comment (cidrblock) [inline ios/l3_interfaces/deleted_example_01.txt]: `encapsulation` should be handled in the l2_interfaces module IMHO

Comment (justjais) [reply ios/l3_interfaces/deleted_example_01.txt]: @cidrblock `trunk encapsulation` is handled in `l2_interfaces`, but I have included the support for `l3_interfaces encapsulation` coz without setting vlan be it dot1q/isl we cannot set the sub-interface ip address.

**Comment by justjais** [comment]
@cidrblock Thanks for giving the comment wrt to secondary IP and I think we can design it as follows, ```

File: ios/l3_interfaces/ios_l3_interface.yaml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+          description:
+          - IPv4 address to be set for the Layer-3 interface mentioned in I(name) option.
+            The address format is <ipv4 address>/<mask>, the mask is number in range 0-32
+            eg. 192.168.0.1/24
+          suboptions:
+            address:
+              description:
+              - Configures the IPv4 address for Interface.
+              type: str
+            secondary:
```
Comment (justjais) [inline ios/l3_interfaces/ios_l3_interface.yaml]: @cidrblock have included support for `secondary` IP and also support for handling ipv4 `dhcp` and ipv6 `autoconfig`.

File: ios/l3_interfaces/ios_l3_interface.yaml [hunk 3]
```diff
@@ -66,24 +66,10 @@ DOCUMENTATION: |
...
-             - Inter Switch Link - Virtual LAN encapsulation, valid vlan identifier range is
-               between 1-1000.
-             type: str
-           priority_tag:
-             description:
-             - Priority-tagged (VLAN 0).
-             type: str
+            dhcp:
+              description:
+              - Obtain a ipv6 address using dhcp.
```
Comment (samccann) [inline ios/l3_interfaces/ios_l3_interface.yaml]: Obtain an IPv6 address using DHCP.

**Pull request merged**